src/modules/flow/iio/nodes.c

```diff
@@ -71,24 +145,82 @@ reader_cb(void *data, struct sol_iio_device *device)
 }
 
 static bool
-create_device_channels(struct gyroscope_data *mdata, int device_id)
+create_device_channels(struct iiodevice_data *mdata, int device_id, enum iiodevice_types type)
 {
     struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
-
+    int i = 0;
+ 
     mdata->device = sol_iio_open(device_id, &mdata->config);
     SOL_NULL_CHECK(mdata->device, false);
 
 #define ADD_CHANNEL(_axis) \
     if (!mdata->use_device_default_scale) \
-        channel_config.scale = mdata->scale._axis; \
+        channel_config.scale = mdata->scales[i]; \
     if (!mdata->use_device_default_offset) \
-        channel_config.offset = mdata->offset._axis; \
-    mdata->channel_ ## _axis = sol_iio_add_channel(mdata->device, "in_anglvel_" # _axis, &channel_config); \
-    SOL_NULL_CHECK_GOTO(mdata->channel_ ## _axis, error);
-
-    ADD_CHANNEL(x);
-    ADD_CHANNEL(y);
-    ADD_CHANNEL(z);
+        channel_config.offset = mdata->offsets[i]; \
+    mdata->channels[i] = sol_iio_add_channel(mdata->device, # _axis, &channel_config); \
+    SOL_NULL_CHECK_GOTO(mdata->channels[i], error);
+
+    switch (type) {
+        case IIO_GYROSCOPE:
+            i = 0;
+            ADD_CHANNEL(in_anglvel_x);
+            i++;
+            ADD_CHANNEL(in_anglvel_y);
+            i++;
+            ADD_CHANNEL(in_anglvel_z);
+            break;
+        case IIO_MAGNETOMETER:
+            i = 0;
+            ADD_CHANNEL(in_magn_x);
+            i++;
+            ADD_CHANNEL(in_magn_y);
+            i++;
+            ADD_CHANNEL(in_magn_z);
+            break;
+        case IIO_ACCELEROMETER:
+            i = 0;
+            ADD_CHANNEL(in_magn_x);
+            i++;
+            ADD_CHANNEL(in_magn_y);
+            i++;
+            ADD_CHANNEL(in_magn_z);
+            break;
+        case IIO_COLOR:
+            i = 0;
+            ADD_CHANNEL(in_intensity_red);
+            i++;
+            ADD_CHANNEL(in_intensity_green);
+            i++;
+            ADD_CHANNEL(in_intensity_blue);            
+            break;
+        case IIO_THERMOMETER:
+            i = 0;
+            ADD_CHANNEL(in_temp);
+            break;
+        case IIO_PRESSURE_SENSOR:
+            i = 0;
+            ADD_CHANNEL(in_pressure);
+            break;            
+        case IIO_HUMIDITY_SENSOR:
+            i = 0;
+            ADD_CHANNEL(in_humidityrelative);
+            break;            
+        case IIO_ADC:
+            i = 0;
+            ADD_CHANNEL(in_voltage0);
+            break;            
+        case IIO_LIGHT_SENSOR:
+            i = 0;
+            ADD_CHANNEL(in_illuminance0);
+            break;            
+        case IIO_PROXIMITY:
+             i = 0;
+            ADD_CHANNEL(in_proximity);
+            break;           
+        default:
+            break;
```

Comment:
You should error here: again, if someone add a new IIO node on future and miss changing here, it should get a nice error.